core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -597,6 +602,13 @@ private[spark] class IndexShuffleBlockResolver(
     }
   }
 
+  override def getBlocksForShuffle(shuffleId: Int, mapId: Long): Seq[BlockId] = {
```

Comment:
In the real data, there is a case where a shuffle index without its data file. So, did we consider those cases in the upper layers where this method is invoked?

Reply:
In that case the external shuffle service will just log unable to delete the file and continue. Is there anywhere to actually know that information?

Reply:
Historically, WorkerDecommission fails with FileNotFound exception because it didn't know about the reality. So, I want to make it sure that the upper layer of this method is safe or not.

Reply:
Do we have a test coverage for that case about the part what you mentioned ?
> In that case the external shuffle service will just log unable to delete the file and continue. Is there anywhere to actually know that information?